jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/resources/LambdaResources.kt

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation.resources
+
+import software.amazon.awssdk.services.cloudformation.CloudFormationClient
+import software.amazon.awssdk.services.cloudformation.model.StackResourceSummary
+import software.amazon.awssdk.services.cloudformation.model.StackSummary
+import software.aws.toolkits.jetbrains.core.CachedResource
+import software.aws.toolkits.jetbrains.core.CachedResourceBase
+
+object CloudFormationResources {
+    private val listStacks: CachedResource<List<StackSummary>> by lazy {
+        CloudFormationCachedResource { listStacksPaginator().stackSummaries().toList() }
+    }
+
+    fun listStacks() = listStacks
+
+    fun listStackResources(stackId: String): CachedResource<List<StackResourceSummary>> =
+        CloudFormationStackResourcesCachedResource(stackId)
+
+    private class CloudFormationCachedResource<T>(private val call: CloudFormationClient.() -> T) :
+        CachedResourceBase<T, CloudFormationClient>(CloudFormationClient::class) {
```

Comment:
In the view iteration this becomes a bit less clunky.

Reply:
Sounds good to me.